apps/users/views.py

```diff
@@ -1,3 +1,178 @@
-from django.shortcuts import render
+import datetime
+
+from drf_util.decorators import serialize_decorator
+from drf_util.utils import gt
+from rest_framework import mixins, status
+from rest_framework.parsers import MultiPartParser
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated, AllowAny
+from rest_framework.response import Response
+from rest_framework.serializers import Serializer
+from rest_framework.viewsets import GenericViewSet, ModelViewSet
+
+from django.shortcuts import get_object_or_404
+
+from apps.common.helpers import generate_code
+from apps.users.models import User, UserVerification, UserAddress
+from apps.users.serializers import UserSerializer, UserVerificationSerializer, UserEmailSerializer, \
+    UserLoginResetSerializer, UserPasswordUpdateSerializer, UserAddressSerializer,\
+    UserRegisterSerializer
+
+from apps.common.permisions import IsAdminOrOwner, IsAdminOrItself, ReadOnly
+
 
 # Create your views here.
+
+
+class UserSignupViewSet(GenericViewSet):
+    """
+    Used for creating and verifying new accounts.
+    """
+    serializer_class = UserRegisterSerializer
+    queryset = User.objects.all()
+    permission_classes = (AllowAny,)
+
+    @serialize_decorator(UserRegisterSerializer)
+    @action(detail=False, methods=['POST'])
+    def register(self, request, *args, **kwargs):
+        raw_password = request.valid.pop('password')
+
+        user, _ = User.objects.get_or_create(**request.valid)
+        if user.is_active:
+            raise ValidationError({'user': 'user with this email exists'})
+
+        user.set_password(raw_password)
+        user.save()
+
+        data = self.get_serializer(user).data
+        return Response(data, status.HTTP_201_CREATED)
+
+    @action(detail=True, methods=['POST'], url_path='send-verification', serializer_class=Serializer)
+    def send_verification(self, request, *args, **kwargs):
+        user = self.get_object()
+
+        if user.is_active:
```

Comment:
ca sa scapi de aceste validari poti seta direct in querysetul view-ului filtru doar dupa userii inactivi